### ISSUE TITLE
Deploy and test use of "guestbook at request" feature on Demo Dataverse

### ISSUE THREAD

**Issue by jggautier** [IQSS/dataverse.harvard.edu#247]
## Overview
After the 6.6 release:
1. Deploy "guestbook at request" feature on demo.dataverse.org
2. The HDV curation team will test the feature for eventual use on dataverse.harvard.edu

Note that sizing only covers deployment on demo; curation testing will occur separately.

**Comment by jggautier** [comment]
[In a Slack conversation on Feb. 20](https://iqss.slack.com/archives/GRDNRJ6BZ/p1708471771340599?thread_ts=1708466471.510619&cid=GRDNRJ6BZ), @sbarbosadataverse wrote about wanting "to confirm it's working exactly as you described"

**Comment by cmbz** [comment]
@jggautier Is the recommendation then to turn on in demo first to confirm, then HDV if the feature behaves as expected?

**Comment by jggautier** [comment]
I'm not sure what @sbarbosadataverse had in mind.

Maybe adding this functionality to Demo Dataverse or maybe to an AWS instance so that @sbarbosadataverse can review it? Or maybe it'll be helpful for @sbarbosadataverse to review documentation related to the functionality and the results of any UX research that was done when it was being worked on?

@sbarbosadataverse what do you think?

**Comment by sbarbosadataverse** [comment]
Yes, let's move to demo and I can actively review and test.

**Comment by jggautier** [comment]
There might be some conversation happening soon [in a thread in the Dataverse Users Community Google Group about using this feature](https://groups.google.com/u/1/g/dataverse-community/c/3M8CHOOnSVU).

**Comment by jggautier** [comment]
Just following up on this after reviewing my conversation with the user who's interested in this feature.

When this is picked up for a sprint and a developer can add this feature to Demo Dataverse, I imagine we'd like it configured so that by default it's turned off for all collections. And collection admins can turn it on [the way Jim described in that Google Groups thread](https://groups.google.com/g/dataverse-community/c/3M8CHOOnSVU/m/o1evZbXoBAAJ).

**Comment by jggautier** [comment]
Just noting here that the GitHub issue at https://github.com/IQSS/dataverse/issues/10625 requests some changes to the way this feature works.

**Comment by jggautier** [comment]
Might be helpful to keep an eye on [a recent Zulip thread](https://dataverse.zulipchat.com/#narrow/stream/378866-troubleshooting/topic/enabling.20guestbook.20at.20request.20for.20instance.20or.20collection) where the manager of another Dataverse installation is asking questions about configuring the "guestbook at request" feature.

**Comment by pdurbin** [comment]
@dliburd and I had a quick chat about this in the context of https://help.hmdc.harvard.edu/Ticket/Display.html?id=369533

I signed up for an account on ADA (where this feature is on) so I could see what the UX is like when you click "Request Access" on this file, for example: https://dataverse.ada.edu.au/file.xhtml?fileId=19166&datasetVersionId=3979

Here are some screenshots:

![Screenshot 2024-10-04 at 4 23 08 PM](https://github.com/user-attachments/assets/a21f0a4d-da04-4c44-8eda-b4ee38d8bcff)

![Screenshot 2024-10-04 at 4 23 15 PM](https://github.com/user-attachments/assets/5a9eeace-daa4-47d8-9a1f-a7ec4e714cb0)

![Screenshot 2024-10-04 at 4 23 24 PM](https://github.com/user-attachments/assets/ef142829-bf23-46b3-a36f-e7039c8ae8a5)

As you can see, the guestbook is shown. Dwayne and I agree that this behavior is quite nice!

As for trying this on demo, judging from https://guides.dataverse.org/en/6.4/installation/config.html#dataverse-files-guestbook-at-request we have a couple options:

- turn "guestbook at request" on globally
- leave "guestbook on request" off globally and selectively turn it on for a few collections for testing

If we go with the latter option, whoever picks up this issue would need to know which collections to turn on "guestbook at request", of course.

**Comment by sbarbosadataverse** [comment]
@kmika11 We plan to move this to demo for testing--fyi--something to keep an eye on since you mentioned users want it in production soon?

**Comment by landreev** [comment]
@jggautier 
The setting `dataverse.files.globus-monitoring-server=true` has been put in place on demo, so the feature can be tested there now. Please let me know if there are problems. (And/or let me know if you want me to remove the setting once the testing is done).

**Comment by qqmyers** [comment]
FWIW: With that setting true, the default changes to using guestbook-at-request. The setting can be changed per collection (on the Edit/General Settings tab) or per dataset (via API) as well. For Harvard Dataverse, one option would be to set the setting false to keep the current default while still allowing guestbook-at-request to be turned on for specific collections.

**Comment by jggautier** [comment]
Thanks. @sbarbosadataverse and @kmika11 are talking about testing.

**Comment by cmbz** [comment]
2025/01/15: Placing on hold until Sonia and Katie complete testing. Julian will confirm with them.

**Comment by kmika11** [comment]
In demo, (presumably with `dataverse.files.guestbook-at-request=true`) there appears to be no way to turn off a guestbook requirement for users to openly/anonymously download files? These are the options I see in the collection edit page: 

<img width="975" alt="Image" src="https://github.com/user-attachments/assets/71ff75d2-2a9e-4c98-8362-33c705a1aa65" />

I think there should _always_ be the option for collection owners to not require any guestbook at any point for their data files. In fact, I strongly think that should be the default for Harvard Dataverse.

**Comment by qqmyers** [comment]
This feature only affects whether a guestbook, if assigned to a dataset, is shown when the user requests data access or when they download. Is there something we should add to the text/help text to make that clearer? Or is there a case you're seeing a guestbook appear when you think it shouldn't? (FWIW: I think a ~guestbook dialog appears if there are custom terms/terms of access in a dataset, intended to assure that users see those terms (either at request access or download time now) before they get the data, even if there is no guestbook assigned to the dataset.)

**Comment by kmika11** [comment]
Ah! I see! I do think the limited info on the create/edit dataverse page and the way the tool tip is worded makes it sound like the only options an author has is between applying a guestbook at download or at request. Perhaps some short text under or with the feature label like "For datasets with enabled guestbooks..." would help clarify that this only applies to specific datasets?

**Comment by kmika11** [comment]
This dataset has no guestbook applied. It has one file with "request access" turned on. 

<img width="830" alt="Image" src="https://github.com/user-attachments/assets/e26fff7c-6dfe-4af7-b2de-75f7074a3d9a" />

When I am logged in as a normal user (i.e. not the creator or a superuser), when I go to request access, I am presented with a guestbook prompt. 

<img width="1261" alt="Image" src="https://github.com/user-attachments/assets/ad30eb32-80bf-4a0a-b556-84eb85b573a0" />

Is this expected? Is there now a standard "guestbook" for all files with "request access" turned on? Compare with this "request access" popup example from Harvard Dataverse that displays only the Terms of Use and does not require any information from the user: 

<img width="1261" alt="Image" src="https://github.com/user-attachments/assets/1fb402ff-6a05-409c-a3cd-e361f654c299" />

**Comment by qqmyers** [comment]
I think that's a bug - I've created a PR [#11203](https://github.com/IQSS/dataverse/pull/11203) to fix that and another issue (terms of access don't appear if you use the per-file download menu in the table (instead of selecting a file and hitting the request access button at the top). I also made minor changes to call the collection/general info/edit option "Guestbook Mode" and to have help that includes "(when configured for a dataset)". Easy to edit that as it gets reviewed.

**Comment by cmbz** [comment]
2025-02-19: Moved to https://github.com/IQSS/dataverse-pm/issues/367

**Comment by cmbz** [comment]
Hi @kmika11 and @sbarbosadataverse Demo has been updated for testing now!

**Comment by cmbz** [comment]
2025-04-15: @kmika11 and @sbarbosadataverse just a reminder to please test/review. Thanks!

**Comment by jggautier** [comment]
@sbarbosadataverse asked me to test/review this feature. I'll be writing my notes in this GitHub comment:

- Add a space between "Request" and "(inherited from enclosing Dataverse)" so that it matches formatting of the option in the similar Curation Labels drop down menu. This is probably just a bug with how that text is formatted.

  <kbd><img width="582" alt="Image" src="https://github.com/user-attachments/assets/136f1f74-fed1-4afb-86f4-91e37752a1cd" /></kbd>

- @kmika11 [wrote in the GitHub issue](https://github.com/IQSS/dataverse.harvard.edu/issues/247#issuecomment-2627507682) that "the create/edit dataverse page and the way the tool tip is worded makes it sound like the only options an author has is between applying a guestbook at download or at request". @qqmyers adjusted the tooltip to try to make it clearer that this setting affects only datasets that have been given a guestbook.
 
  Why don't we let collection managers change the "guestbook mode" from the Manage Guestbook page instead, where they're already able to create, edit, and download guestbooks?
  - This might help us avoid this confusion.
  - This might also make it easier for collection managers to see how their guestbook responses have been collected.
  - Putting this information on the same page might also make it easier for the user to know that this setting affects the behavior of all guestbooks in their collection but may not affect guestbooks "inherited" from that collection's parent collection.

- If this hasn't been done already, I recommend that someone talk with managers of repositories that have already turned this setting on, especially repositories that let users create collections, as opposed to a handful of repository staff, to see how well it's been working and to help evaluate the seriousness or prevalence of the challenges we've already discussed.

- Similarly, [I wrote earlier in this GitHub issue](https://github.com/IQSS/dataverse.harvard.edu/issues/247#issuecomment-1992089690) that someone using Harvard Dataverse was interested in being able to do what this feature might let them do. Following up with that person about this feature might also help us learn how well it's working and discover other challenges with how it's designed.

**Comment by kmika11** [comment]
The guestbook at request appears to work as expected for me now!

I do agree with @jggautier that it would make more sense as a collection manager to be able to choose a guestbook's behavior in the "manage guestbook" page rather than the collection's "general info" page. Could there be scenarios where one collection may want datasets with different guestbook settings applied?

**Comment by cmbz** [comment]
2025-04-21
- Discussed during prio meeting
- @sbarbosadataverse will review comments and make a recommendation for moving it on into production

**Comment by sbarbosadataverse** [comment]
@jggautier Thanks for the deep dive into this issue, although all we needed at the moment was to ensure the bugs we recently fixed are indeed fixed. I am capturing anything from your summary that appears to be a bug versus future improvements to the feature. Please confirm that nothing is preventing you from using the feature as it is now? 

@cmbz one small fix to the feature and then we should move this forward to release for hdv. And we can add Julian's comments above, to a feature enhancement issue for further conversation.

**Bugs to fix before the release of this feature:**
1. Add a space between "Request" and "(Default)" and between "Request" and "(inherited from enclosing Dataverse)" so that it matches formatting of the option in the similar Curation Labels drop down menu on the page. This is probably just a bug with how that text is formatted.


@kmika11 Just to confirm you didn't run into any further bugs in your testing?

**Comment by qqmyers** [comment]
This can be temporarily fixed in v6.6 by adding a space to the end of https://github.com/IQSS/dataverse/blob/399cf9bf9abd939e6c5cdf1383dbab5327f42cd5/src/main/java/propertyFiles/Bundle.properties#L1011-L1012 .  To keep it easier to maintain and translate, rather than having an ~invisible space at the end of these properties, I've added the space programmatically in https://github.com/IQSS/dataverse/pull/11445.

**Comment by cmbz** [comment]
2025-04-28
- @sbarbosadataverse will speak with @kmika11 about timing, whether this deployment can wait for Release 6.7.

**Comment by cmbz** [comment]
2025-05-5
- @cmbz pinged @kmika11 about the [above question](https://github.com/IQSS/dataverse.harvard.edu/issues/247#issuecomment-2835462409) in Slack

**Comment by cmbz** [comment]
2025-05-05
- Closing as complete. 
- Please see follow-up issues: 
  - https://github.com/IQSS/dataverse.harvard.edu/issues/366
  - https://github.com/IQSS/dataverse/issues/11468
